src/docs/development/tools/devtools/code-size.md

```diff
@@ -0,0 +1,194 @@
+`package:b` and `package:c`, instead of its "dominator", `package:a`.
+
+```
+package:a --> package:b -->
+                              package:d
+package:a --> package:c -->
+```
+
+## Diff Tab
+
+![Screenshot of code size diff]({% asset tools/devtools/code_size_diff.png @path %}){:width="100%"}
+
+### Loading code size files
+
+When you open the Diff tab, you'll see instructions to load "old" and "new" code
+size files. These files should be generated from two different versions of the
+same app, for example, the bits before and after a PR. Drag and drop these files
+in to their respective locations, and click "Analyze Diff".
+
+![Screenshot of code size diff loading screen]({% asset tools/devtools/code_size_load_diff.png @path %}){:width="100%"}
+
+See [Generating size files][] below for information on generating these files.
+
+### Treemap and table
+
+In the diff view, the treemap and tree table show only data that differs between
+the two imported size files.
+
+For questions about using the treemap, see [How to use the treemap][] below.
+
+## How to use the treemap
+
+A treemap is a visualization for hierarchical data. The space is broken up into
+rectangles, where each rectangle is sized and ordered by some quantitative
+variable (in this case, size in bytes). Inside of each rectangle (call one A),
+there are additional rectangles that exist one level deeper in the data
+hierarchy (children of A).
+
+To drill into a cell in the treemap, select the cell. This will re-root the tree
+to the selected cell.
+
+To navigate back, or up a level, use the breadcrumb navigator at the top of the
+treemap.
+
+![Screenshot of treemap breadcrumb navigator]({% asset tools/devtools/treemap_breadcrumbs.png @path %}){:width="100%"}
+
+## Generating size files
+
+### Flutter code size analysis files
+
+A prerequisite to using the code size tool is generating the proper data to
+analyze. To analyze the size of your entire app, including analysis for Native
+and Dart code, generate a code size analysis file using the `--analyze-size`
+flag:
+
+```
+flutter build <your target platform> --analyze-size
+```
+
+This will build your application, print a size summary to the command line, and
+print a line telling you where to find the size analysis file.
+
+```
+flutter build apk --analyze-size --target-platform=android-arm64
+...
+▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒
+app-release.apk (total compressed)                               6 MB
+...
+▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒
+A summary of your APK analysis can be found at: build/apk-code-size-analysis_01.json
+```
+
+In this example, the file found at `build/apk-code-size-analysis_01.json` is the
+file we will import into the DevTools Code Size Tools to analyze further.
+
+### Dart AOT snapshot files
+
+If you are not interested in total app size, but rather are only interested in
```

Comment:
=>
"If you are interested in Dart code size (rather than total app size), you can also import Dart AOT v8 snapshots.
You can generate these files as follows:"
(Don't use "via".)